crates/tauri-utils/src/acl/build.rs

```diff
@@ -460,9 +455,23 @@ pub fn generate_allowed_commands(
     })
     .collect();
 
-  capabilities.extend(crate::acl::get_capabilities(&config, None, None)?);
+  let capabilities_file_path = out_dir.join(CAPABILITIES_FILE_NAME);
+  // the capabilities file only exist in the tauri-build context
+  if !capabilities_file_path.exists() {
```

Comment:
Would this call be a problem when the build script re-runs? I think maybe we could move back the 

```rust
  if let Some(capabilities_file_path) = pre_built_capabilities_file_path {
    if capabilities_file_path.exists() {
      let capabilities_file =
        std::fs::read_to_string(capabilities_file_path).context("failed to read capabilities")?;
      capabilities_from_files =
        serde_json::from_str(&capabilities_file).context("failed to parse capabilities")?;
    }
  }
```

part to `tauri-codegen`, and replace `pre_built_capabilities_file_path` param with `capabilities_from_files: BTreeMap<String, Capability>`

Reply:
same, it was a nightmare 🤣